coalib/testing/LocalBearTestHelper.py

```diff
@@ -157,18 +173,55 @@ def check_results(self,
             if check_order:
                 msg += ' Or the order may be wrong.'
 
-        with prepare_file(lines, filename,
-                          force_linebreaks=force_linebreaks,
-                          create_tempfile=create_tempfile,
-                          tempfile_kwargs=tempfile_kwargs) as (file, fname), \
-                execute_bear(local_bear, fname, file,
-                             **settings) as bear_output:
-            if not check_order:
-                self.assertEqual(sorted(bear_output), sorted(results), msg=msg)
-            else:
-                self.assertEqual(bear_output, results, msg=msg)
-
-            return bear_output
+        bear_output = get_results(local_bear, lines,
+                                  filename=filename,
+                                  force_linebreaks=force_linebreaks,
+                                  create_tempfile=create_tempfile,
+                                  tempfile_kwargs=tempfile_kwargs,
+                                  settings=settings)
+        if not check_order:
+            self.assertEqual(sorted(bear_output), sorted(results), msg=msg)
+        else:
+            self.assertEqual(bear_output, results, msg=msg)
+
+        return bear_output
+
+    def check_line_result_count(self,
+                                local_bear,
+                                lines,
+                                results_num,
+                                filename=None,
+                                check_order=False,
+                                force_linebreaks=True,
+                                create_tempfile=True,
+                                tempfile_kwargs={},
+                                settings={}):
+        """
+        Check many results for each line.
+
+        :param local_bear:       The local bear to check with.
+        :param lines:            The lines to check. (List of strings)
+        :param results_num:      The expected list of many results each line.
+        :param filename:         The filename, if it matters.
+        :param force_linebreaks: Whether to append newlines at each line
+                                 if needed. (Bears expect a \\n for every line)
+        :param create_tempfile:  Whether to save lines in tempfile if needed.
+        :param tempfile_kwargs:  Kwargs passed to tempfile.mkstemp().
+        :param settings:         A dictionary of keys and values (both strings)
+                                 from which settings will be created that will
+                                 be made available for the tested bear.
+        """
+        for line, num in zip(lines, results_num):
+            if line.strip() == '':
```

Comment:
ugh.  this consumes a number from `results_num`

That means doing inside the `zip` doesnt work.

Instead, before the `for` loop, modify `lines` to remove blank lines and lines beginning with `#`.